web/pdf_attachment_viewer.js

```diff
@@ -89,6 +89,43 @@ var PDFAttachmentViewer = (function PDFAttachmentViewerClosure() {
     /**
      * @private
      */
+    _bindPdfLink:
+        function PDFAttachmentViewer_bindPdfLink(button, content, filename) {
+      var blobUrl;
+      button.onclick = function() {
+        if (!blobUrl) {
+          blobUrl = pdfjsLib.createObjectURL(
+            content, 'application/pdf', pdfjsLib.PDFJS.disableCreateObjectURL);
+        }
+        var viewerUrl;
+        if (typeof PDFJSDev === 'undefined' || PDFJSDev.test('GENERIC')) {
+          // The current URL is the viewer, let's use it and append the file.
+          viewerUrl = '?file=' + encodeURIComponent(blobUrl + '#' + filename);
+        } else if (PDFJSDev.test('CHROME')) {
+          // In the Chrome extension, the URL is rewritten using the history API
+          // in viewer.js, so an absolute URL must be generated.
+          // eslint-disable-next-line no-undef
+          viewerUrl = chrome.runtime.getURL('/content/web/viewer.html') +
+            '?file=' + encodeURIComponent(blobUrl + '#' + filename);
+        } else {
```

Comment:
Do we actually need to use such complicated logic below, since this code block really only ought to be relevant for `FIREFOX`/`MOZCENTRAL` builds, where `URL.createObjectURL` should always be supported anyway?

I.e. wouldn't it be sufficient to do something like the below instead, and just ignore the `pdfjsLib.PDFJS.disableCreateObjectURL === true` case; given that there's no way for a user of the addon/built-in version to disable `createObjectURL` short of editing the source code.
```javascript
} else if (PDFJSDev.test('FIREFOX || MOZCENTRAL')) {
  // Let Firefox's content handler catch the URL and display the PDF.
  viewerUrl = blobUrl + '?' + encodeURIComponent(filename);
}
```